plugins/modules/postgresql_pg_hba.py

```diff
@@ -305,6 +312,126 @@ class PgHbaRuleValueError(PgHbaRuleError):
     '''
 
 
+class TokenizerException(Exception):
+    """
+    This exception is raised when a string can't be tokenized
+    """
+
+
+def parse_hba_file(input_string):
+    """
+    This function parses a complete pg_hba.conf file into a list of tuples where each tuple represents a rule in the
+    file.
+    :param input_string: The whole string in the pg_hba.conf file (not just a single line)
+    :return: A list of Rule objects that represents the contents of the input string.
+    """
+    rules = []
+    line_iter = iter(input_string.split("\n"))
+    line = next(line_iter, None)
+    while line is not None:
+        # if that line continues, we just glue the next line onto the end until it ends
+        # we can and have to do that, as continuation even applies within comments and quoted strings [sic]
+        # https://www.postgresql.org/docs/current/auth-pg-hba-conf.html#AUTH-PG-HBA-CONF
+        comment = None
+        while line.endswith("\\"):
+            cont_line = next(line_iter, None)
+            if cont_line is None:
+                # we got a line continuation, but there was no more line
+                raise PgHbaRuleError("The last line ended with a '\\' (line continuation).")
+            line += "\n" + cont_line  # add the newline so we don't lose that information
```

Comment:
I actually mean ["lose" (verb)](https://dictionary.cambridge.org/dictionary/english/lose) as in "to no longer have something because you do not know where it is", not "loose" (verb) as in making something (a knot or nut) loose